server/src/main/java/org/elasticsearch/index/search/stats/FieldUsageStats.java

```diff
@@ -0,0 +1,279 @@
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.field(TOTAL, getTotal());
+            builder.field(TERMS, terms);
+            builder.field(PROXIMITY, getProximity());
+            builder.field(STORED_FIELDS, storedFields);
+            builder.field(DOC_VALUES, docValues);
+            builder.field(POINTS, points);
+            builder.field(NORMS, norms);
+            builder.field(TERM_VECTORS, termVectors);
+            builder.field(FREQS, freqs);
+            builder.field(POSITIONS, positions);
+            builder.field(OFFSETS, offsets);
+            builder.field(PAYLOADS, payloads);
+            return builder;
+        }
+
+        public Set<UsageContext> keySet() {
+            final EnumSet<UsageContext> set = EnumSet.noneOf(UsageContext.class);
+            if (terms > 0L) {
+                set.add(UsageContext.TERMS);
+            }
+            if (freqs > 0L) {
+                set.add(UsageContext.FREQS);
+            }
+            if (positions > 0L) {
+                set.add(UsageContext.POSITIONS);
+            }
+            if (offsets > 0L) {
+                set.add(UsageContext.OFFSETS);
+            }
+            if (docValues > 0L) {
+                set.add(UsageContext.DOC_VALUES);
+            }
+            if (storedFields > 0L) {
+                set.add(UsageContext.STORED_FIELDS);
+            }
+            if (norms > 0L) {
+                set.add(UsageContext.NORMS);
+            }
+            if (payloads > 0L) {
+                set.add(UsageContext.PAYLOADS);
+            }
+            if (termVectors > 0L) {
+                set.add(UsageContext.TERM_VECTORS);
+            }
+            if (points > 0L) {
+                set.add(UsageContext.POINTS);
+            }
+            return set;
+        }
+
+        public long getTerms() {
+            return terms;
+        }
+
+        public long getFreqs() {
+            return freqs;
+        }
+
+        public long getPositions() {
+            return positions;
+        }
+
+        public long getOffsets() {
+            return offsets;
+        }
+
+        public long getDocValues() {
+            return docValues;
+        }
+
+        public long getStoredFields() {
+            return storedFields;
+        }
+
+        public long getNorms() {
+            return norms;
+        }
+
+        public long getPayloads() {
+            return payloads;
+        }
+
+        public long getTermVectors() {
+            return termVectors;
+        }
+
+        public long getPoints() {
+            return points;
+        }
+
+        public long getProximity() {
+            return freqs + offsets + positions + payloads;
+        }
+
+        public long getTotal() {
+            return terms + freqs + positions + offsets + docValues + storedFields + norms + payloads + termVectors + points;
+        }
```

Comment:
likewise could we track the total separately so that it only increments by one regardless of how the field is accessed?

Reply:
"total" by its very name feels to me like a sum of the other fields (and could be confusing if it was not). I can understand why you would like to count "access to any part of a field" as just one event, I'm just not sure that it should carry the name "total" (especially as in the disk usage PR it's a sum). Do you have a suggestion for a different name? Or are you less concerned about this?

Reply:
I agree that `total` wouldn't be a good name in that case. I'm thinking of "any" right now.